include/seqan3/core/metafunction/range.hpp

```diff
@@ -180,20 +190,20 @@ struct size_type<rng_t>
  * Attention, this metafunction implicitly removes cv-qualifiers on the all value_types except the one returned.
  */
 template <typename t>
-//!\cond
-    requires requires (t) { typename value_type_t<remove_cvref_t<t>>; }
-//!\endcond
-struct innermost_value_type
+struct innermost_value_type;
+
+template <typename t>
+    requires detail::has_value_type<t>
+struct innermost_value_type<t>
 {
     //!\brief The return type (recursion not shown).
     using type = value_type_t<remove_cvref_t<t>>;
 };
 
 //!\cond
 template <typename t>
-    requires requires (t)
+    requires detail::has_value_type<t> && requires
```

Comment:
why not change this to just
```cpp
requires detail::has_value_type<t> && detail::has_value_type<value_type_t<t>>
```
Or does this fail because of GCC not lazy instantiating? Then you could still do:
```cpp
requires detail::has_value_type<t> && detail::has_value_type<detail::type_trait_or_t<value_type<t>, void>
```